File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.17-pattern.js
function log(val) {
    process.stdout.write(String(val));
}

function pattern_17() {
    let chars = ['a', 'b', 'c', 'd', 'e'];

    let row = 5;
    let column = 5;

    for (let y = 0; y < row; y++) {
        for (let x = 0; x < column; x++) {
            if (x <= y) {
                log(chars[y]);
            } else {
                log(' ');
            }
        }

        console.log();
    }
}

pattern_17();

File: python/0.0-Basics/0.3-build-up-logical-thinking/0.3.17-pattern.py
from utils.clear import clear


def pattern_17():
    row = 5
    column = 5

    chars = ['a', 'b', 'c', 'd', 'e']

    for i in range(0, row ):
        for j in range(0, column):
            if j <= i:
                print(chars[i], end="")
            elif j > i:
                print(" ", end="")

        print()


clear()
pattern_17()
